test/e2e/apps/daemon_set.go

```diff
@@ -801,6 +808,142 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
+				found := ds.ObjectMeta.Name == testDaemonset.ObjectMeta.Name &&
+					ds.ObjectMeta.Namespace == testDaemonset.ObjectMeta.Namespace &&
+					ds.Labels[daemonsetNameLabel] == dsName
+				if !found {
+					framework.Logf("Observed daemon set %v in namespace %v with annotations: %v & Conditions: %v", ds.ObjectMeta.Name, ds.ObjectMeta.Namespace, ds.Annotations, ds.Status.Conditions)
+					return false, nil
+				}
+				for _, cond := range ds.Status.Conditions {
+					if cond.Type == "StatusUpdate" &&
+						cond.Reason == "E2E" &&
+						cond.Message == "Set from e2e test" {
+						framework.Logf("Found daemon set %v in namespace %v with labels: %v annotations: %v & Conditions: %v", ds.ObjectMeta.Name, ds.ObjectMeta.Namespace, ds.ObjectMeta.Labels, ds.Annotations, ds.Status.Conditions)
+						return found, nil
+					} else {
+						framework.Logf("Observed daemon set %v in namespace %v with annotations: %v & Conditions: %v", ds.ObjectMeta.Name, ds.ObjectMeta.Namespace, ds.Annotations, ds.Status.Conditions)
+						return false, nil
+					}
+				}
+			}
+			framework.Logf("Observed event: %+v", event.Type)
+			return false, nil
+		})
+		framework.ExpectNoError(err, "failed to locate daemon set %v in namespace %v", testDaemonset.ObjectMeta.Name, ns)
+		framework.Logf("Daemon set %s has an updated status", dsName)
+
+		ginkgo.By("patching the DaemonSet Status")
+		daemonSetStatusPatch := appsv1.DaemonSet{
+			Status: appsv1.DaemonSetStatus{
+				Conditions: []appsv1.DaemonSetCondition{
+					{
+						Type:   "StatusPatched",
+						Status: "True",
+					},
+				},
+			},
+		}
+
+		payload, err := json.Marshal(daemonSetStatusPatch)
+		framework.ExpectNoError(err, "Failed to marshal JSON. %v", err)
+		_, err = dsClient.Patch(context.TODO(), dsName, types.MergePatchType, payload, metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err, "Failed to patch daemon set status", err)
+
+		ginkgo.By("watching for the daemon set status to be patched")
+		ctx, cancel = context.WithTimeout(context.Background(), dsRetryTimeout)
+		defer cancel()
+		_, err = watchtools.Until(ctx, dsList.ResourceVersion, w, func(event watch.Event) (bool, error) {
+			if ds, ok := event.Object.(*appsv1.DaemonSet); ok {
+				found := ds.ObjectMeta.Name == testDaemonset.ObjectMeta.Name &&
+					ds.ObjectMeta.Namespace == testDaemonset.ObjectMeta.Namespace &&
+					ds.Labels[daemonsetNameLabel] == dsName
+				if !found {
+					framework.Logf("Observed daemon set %v in namespace %v with annotations: %v & Conditions: %v", ds.ObjectMeta.Name, ds.ObjectMeta.Namespace, ds.Annotations, ds.Status.Conditions)
+					return false, nil
+				}
+				for _, cond := range ds.Status.Conditions {
+					if cond.Type == "StatusPatched" {
+						framework.Logf("Found daemon set %v in namespace %v with labels: %v annotations: %v & Conditions: %v", ds.ObjectMeta.Name, ds.ObjectMeta.Namespace, ds.ObjectMeta.Labels, ds.Annotations, ds.Status.Conditions)
+						return found, nil
+					} else {
```

Comment:
minor style thing, you don't need an else here.  If you "return" / "continue" / "break" in an if, you should exclude the else for clarity.

Reply:
I did try that but found [verify-staticcheck.sh](https://github.com/kubernetes/kubernetes/blob/master/hack/verify-staticcheck.sh) fails with `the surrounding loop is unconditionally terminated (SA4004)`. Using `else` was my way of dealing with that error.

Reply:
I don't think the static check would be right in that case anyway.  Static check doesn't trump style.

Reply:
I updated this yesterday. The else block has been removed.